federation/pkg/federation-controller/configmap/configmap_controller_test.go

```diff
@@ -143,11 +143,18 @@ func TestConfigMapController(t *testing.T) {
 	}
 
 	configmapWatch.Modify(configmap1)
-	updatedConfigMap = GetConfigMapFromChan(cluster1UpdateChan)
-	assert.NotNil(t, updatedConfigMap)
-	assert.Equal(t, configmap1.Name, updatedConfigMap.Name)
-	assert.Equal(t, configmap1.Namespace, updatedConfigMap.Namespace)
-	assert.True(t, util.ConfigMapEquivalent(configmap1, updatedConfigMap))
+	for {
```

Comment:
That's reasonable. It still seems like this is an issue that could happen in the real world, but I suppose that's OK because it's still eventually consistent.